main.tf

```diff
@@ -503,3 +503,24 @@ resource "aws_iam_role_policy_attachment" "eip" {
   role       = aws_iam_role.instance.name
   policy_arn = aws_iam_policy.eip[0].arn
 }
+
+################################################################################
+### Lambda function for ASG instance termination lifecycle hook
+################################################################################
+module "terminate_workers_lifecycle_function" {
+  source = "./modules/terminate-workers"
+
+  count = var.asg_terminate_lifecycle_hook_create ? 1 : 0
```

Comment:
question: Should we really make this configurable? As far as I understand this PR (and also from my daily work with this module) it fixes a bug. Thus it should always be enabled to make sure that there are no dangling instances. @npalm

Reply:
Yes I would suggest to make it configurable, but with default enabled. We don't know who is using the module but some orgs are very restrictive for lambda, for example enforcing using lambda's in a custom VPC. By making the the feature optiona (opt-out) we do not block anyone to use the module

Reply:
Pushed an update to enable this by default